src/engine/init.c

```diff
@@ -60,7 +60,13 @@ const char	       *dss_socket_dir = "/var/run/daos_server";
 int			dss_nvme_shm_id = DAOS_NVME_SHMID_NONE;
 
 /** NVMe mem_size for SPDK memory allocation when using primary mode */
-int			dss_nvme_mem_size = DAOS_NVME_MEM_PRIMARY;
+int			dss_nvme_mem_size;
+
+/** NVMe nr_hugepages defined in server config */
+int			dss_nvme_nr_hugepages;
```

Comment:
Rather than having two variables, one for number of pages and one for size , I would just have a single huge_page_total_mem (just an example name) to specify the total number of huge page bytes available. I don't think the differentiation between number of pages and size of each page is relevant in the engine.